lib/charms/hpc_libs/v0/slurm_ops.py

```diff
@@ -64,17 +69,34 @@ def _on_install(self, _) -> None:
 import os
 import socket
 import subprocess
+import textwrap
 from abc import ABC, abstractmethod
 from collections.abc import Mapping
 from contextlib import contextmanager
 from enum import Enum
 from pathlib import Path
 from typing import Any, Optional, Union
 
+import distro
 import dotenv
 import yaml
 from slurmutils.editors import slurmconfig, slurmdbdconfig
 
+
+class SlurmOpsError(Exception):
+    """Exception raised when a slurm operation failed."""
+
+    @property
+    def message(self) -> str:
+        """Return message passed as argument to exception."""
+        return self.args[0]
+
+
+try:
+    import charms.operator_libs_linux.v0.apt as apt
+except ImportError as e:
+    raise SlurmOpsError(f"could not import apt charm library. reason: {e}")
```

Comment:
Should we raise `SlurmOpsError` here if we fail to load the the apt charm library, or should we just "bubble up" the `ImportError` and have an error log message that the apt charm library isn't available and how to correct the issue? `ImportError` is still valid here as we're having "troubles" importing library dependencies.

```python3
try:
    import charms.operator_libs_linux.v0.apt as apt
except ImportError as e:
    _logger.error("down bad out here...")
   raise
```

Also, this way we don't have to define `SlurmOpsError` error before importing the apt charm library. _We'll just define the logger first instead_. Let me know what your thoughts are here.

Reply:
I don't think we should depend on `_logger` for error reporting, since people can just disable it globally.

Reply:
If the problem is using `SlurmOpsError`, I can just use `ImportError` instead

Reply:
Looking at the docs for [`ImportError`](https://docs.python.org/3/library/exceptions.html#ImportError), looks like we can only set the name of the module that we failed to import and the path to the module that's causing the problems. 

Given that we'd need to hack `ImportError` to include a message for how to install the `apt` charm library, let's just use `SlurmOpsError` here even if it feels funky to declare a custom exception before an import.

Reply:
_Ah wait_, it looks like you can pass a message to `ImportError`